discovery-provider/plugins/notifications/src/processNotifications/mappers/repost.ts

```diff
@@ -34,30 +34,55 @@ export class Repost extends BaseNotification<RepostNotificationRow> {
       return acc
     }, {} as Record<number, { name: string, isDeactivated: boolean }>)
 
-
     if (users?.[this.receiverUserId]?.isDeactivated) {
       return
     }
 
-    // TODO: Fetch the tracks or playlist
-
-    // Get the user's notification setting from identity service
     const userNotifications = await super.getShouldSendNotification(this.receiverUserId)
+    const reposterUserName = users[this.repostUserId]?.name
+    let entityType
+    let entityName
+
+    if (this.repostType === EntityType.Track) {
+      const res: Array<{ track_id: number, title: string }> = await this.dnDB.select('track_id', 'title')
+        .from<TrackRow>('tracks')
+        .where('is_current', true)
+        .whereIn('track_id', [this.repostItemId])
+      const tracks = res.reduce((acc, track) => {
+        acc[track.track_id] = { title: track.title }
+        return acc
+      }, {} as Record<number, { title: string }>)
+
+      entityType = 'track'
+      entityName = tracks[this.repostItemId]?.title
+    } else {
+      const res: Array<{ playlist_id: number, playlist_name: string, is_album: boolean }> = await this.dnDB.select('playlist_id', 'playlist_name', 'is_album')
+        .from<PlaylistRow>('playlists')
+        .where('is_current', true)
+        .whereIn('playlist_id', [this.repostItemId])
+      const playlists = res.reduce((acc, playlist) => {
+        acc[playlist.playlist_id] = { playlist_name: playlist.playlist_name, is_album: playlist.is_album }
+        return acc
+      }, {} as Record<number, { playlist_name: string, is_album: boolean }>)
+      const playlist = playlists[this.repostItemId]
+      entityType = playlist?.is_album ? 'album' : 'playlist'
+      entityName = playlist?.playlist_name
```

Comment:
should there be a default `entityName` to fallback on in case playlist doesn't exist? same for the save notification

Reply:
In that case it should just error - will make a followup task to handle and log errors so we know, thx